guides/tutorials/mediation.md

```diff
@@ -0,0 +1,120 @@
+    Alice->>Bob: WaitForConnection
+
+    Alice->>Bob: SendMessage Alice->Bob
+    Bob->>Mediator: SendMessage Bob->Alice (Mediator)
+    Mediator->>Alice: ForwardMessage Bob->Alice (Mediator)
+```
+
+### What is a mediator, and Why should I use one?
+
+A mediator is just a special type of Agent.
+
+:::info
+"When we describe agent-to-agent communication, it is convenient to think of an interaction only in terms of Alice and Bob and their agents. We say things like: "Alice's agent sends a message to Bob's agent" -- or perhaps "Alice's edge agent sends a message to Bob's cloud agent, which forwards it to Bob's edge agent".
+
+Such statements adopt a useful level of abstraction--one that's highly recommended for most discussions. However, they make a number of simplifications. By modeling the roles of mediators and relays in routing, we can support routes that use multiple transports, routes that are not fully known (or knowable) to the sender, routes that pass through mix networks, and other advanced and powerful concepts."[Aries RFC 0046: Mediators and Relays](https://github.com/hyperledger/aries-rfcs/blob/main/concepts/0046-mediators-and-relays/README.md#aries-rfc-0046-mediators-and-relays)
+
+:::
+
+A practical example where the use of mediators is beneficial can be found in a mobile environment. A mobile agent, aka. your mobile wallet, receives messages from other agents. Technically, sending HTTP messages to a mobile agent is impossible because the agent does not have a public IP. As you have probably seen in previous examples, a workaround for this scenario is to use ngrok. Another option is to use a mediator. When an agent is configured to use a mediator, messages to that agent are routed through the mediator and from there to its destination. Thus, a mobile agent can use a "public" mediator to receive messages on its behalf and poll them from time to time. Here WebSockets come in handy because instead of polling messages from the mediator, the mediator can use WebSockets to push messages to the mobile agent. This is just one example of how a mediator can be used. Further resources about mediators can be found [here](./mediation.md#useful-resources).
+
+### 1. Agent Configuration
+
+First, we need to setup the agents.
+Depending on your target, [React
+Native](../getting-started/installation/react-native) or
+[Node.js](../getting-started/installation/react-native), this might vary.
+
+This tutorial assumes a [Node.js environment](../getting-started/installation/nodejs) for all agents (_Alice_, _Bob_, _Meditator_).
+
+To set up the agents, we use the `startAgent(...)` utility function. The function takes an object as argument containing the `name`, `port`, and `modules` of the agent to be created. In this example, the `modules` property is used to configure the `RecipientModule`, which is required to set up a mediator for an agent. Therefore, only Alice must provide this property.
+
+```typescript showLineNumbers mediation.ts section-1
```

Comment:
This is still present in the other tutorials, but we're stating to look more at not using e.g. `startAgents` methods as it make the code example more complex and less aligned with how you would set this up in real life. 

I think having either separate flows for setting up an recipient agent, the mediator etc.., and then having multilpe files that set-up the mediator, recipient, etc.. agents. I think it's nice to have a single file that runs all three in a single file, but people often just copy and paste code.

I think having two separate tutorials would work great here:
1. Set up an agent to receive message through a mediator. We could use animo's mediator url as an example, or we can point to the set up a mediator tutorial if you want to set it up yourself.
2. Set up a simple mediator agent to route messages for clients. We can point users to the toolbox and use that as the recipient agent, while still using the AFJ mediator.

This means we focus more on how to set up these components in isolations. Curious to hear your thought. This article is really great as it stands, so this could also be a future addition. It's just something we've noticed over the last months with the existing tutorials.

Reply:
I completely agree. I have not done that because I did not want to bloat the example with "repetitive setup code", as the complete example is already 120+ loc. What do you think?

"""I think having two separate tutorials would work great here: ... """
Yes, I also think that would work out nicely here. Maybe we can even have three tutorials. Considering this right now, the most basic e2e example, where we also explain the basic concepts, whereas in your 1. we can show how to set up a recipient when you are not in control of the mediator, and you need to establish a connection, request mediation, and await the grant. This would also cover the case you asked me about during the AFJ call.
I'll leave it up to you to decide if you consider this a future addition or if you want to do it now.

Reply:
I'm unsure what was asked during the AFJ WG last week, but I agree with separating things out a bit, maybe even on separate pages.